e2e/e2e_test.go

```diff
@@ -99,6 +99,50 @@ func TestE2EBasic(t *testing.T) {
 	_ = cleanup()
 }
 
+// TestE2EArgumentParsing runs the CTS binary and checks to ensure that
+// argument parsing works properly for both `=` and space-delimited argument values.
+// For example, both `--a=b` and `--a b` should be acceptable ways to specify args.
+func TestE2EArgumentParsing(t *testing.T) {
+	srv := newTestConsulServer(t)
+	defer srv.Stop()
+
+	tempDir := fmt.Sprintf("%s%s", tempDirPrefix, "cmd_parsing")
+	cleanupTemp := testutils.MakeTempDir(t, tempDir)
+
+	// Create an extra, empty directory that will be searched for config files.
+	// This is done because the `StartCTS` function always includes its own
```

Comment:
I'm leaving the above comment so that you're aware that `runSubcommand` exists, but I'm realising that it is a blocking call, and removes the ability to check if CTS started up correctly. Not a great fit for this particular scenario, so your workaround is the better option.